v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/models/PreparedStatementValueObject.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.models;
+
+public class PreparedStatementValueObject<T> {
+  private String dataType;
+  private T value;
+
+  public PreparedStatementValueObject(String dataType, T value) {
+    this.dataType = dataType;
+    this.value = value;
+  }
+
+  public String getDataType() {
+    return dataType;
+  }
+
+  public void setDataType(String dataType) {
+    this.dataType = dataType;
+  }
+
+  public T getValue() {
+    return value;
+  }
+
+  public void setValue(T value) {
+    this.value = value;
+  }
+
+  // Override toString() for better readability when printing objects
+  @Override
+  public String toString() {
+    return "PreparedStatementValueObject{" + "key='" + dataType + '\'' + ", value=" + value + '}';
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) return true;
+    if (obj == null || getClass() != obj.getClass()) return false;
```

Comment:
If we consider using [com.google.common.base.Objects.equal](https://guava.dev/releases/19.0/api/docs/com/google/common/base/Objects.html), it would do internally do all the nullability checks for you.

Reply:
In case you use AutoValue [as we have discussed here](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/2072/files#r1891134130), a default implementation of equals comes out of box for you.

Reply:
Sure @VardhanThigle